packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -167,30 +213,42 @@ exports.sourceNodes = async (
         })
       }
 
-      const entryNode = {
+      let entryNode = {
         id: entryItem.sys.id,
         parent: contentTypeItemId,
         children: [],
-        ...entryItemFields,
         internal: {
           type: `${makeTypeName(contentTypeItemId)}`,
-          content: JSON.stringify(entryItem),
           mediaType: `application/json`,
         },
       }
 
+      // Replace text fields with text nodes so we can process their markdown
```

Comment:
Yeah, doing markdown conversion at compile time is much nicer. Much less client code + somewhat faster render times.